pkg/controller/main-controller.go

```diff
@@ -428,8 +441,12 @@ func leaderRun(ctx context.Context, c *Controller, threadiness int, stopCh <-cha
 	// 2) we need to make sure we have STS API certificates (if enabled)
 	if IsSTSEnabled() {
 		go func() {
-			klog.Infof("STS is enabled, starting STS API certificate setup")
-			c.generateSTSTLSCert()
+			if utils.GetOperatorRuntime() == common.OperatorRuntimeOpenshift {
+				klog.Infof("STS is enabled, skipping TLS certificate generation")
```

Comment:
Should explain reason for skipping cert generation.

Reply:
applied suggestion